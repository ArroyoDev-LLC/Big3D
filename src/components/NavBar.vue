<script lang="ts">
import { defineComponent, ref } from 'vue'
import Menu from 'primevue/menu'
import { breakpointsTailwind, useBreakpoints } from '@vueuse/core'
const breakpoints = useBreakpoints(breakpointsTailwind)
const sm = breakpoints.smaller('sm')

interface NavItem {
  label: string
  command: Function
}

export default defineComponent({
  name: 'NavBar',
  components: { Menu },
  props: {
    isUserLoggedIn: {
      type: Boolean,
      required: true
    },
    navItems: {
      type: Array<NavItem>,
      default: []
    }
  },
  setup(props, { emit }) {
    const tabClick = (item) => {
      item.command()
      emit('tabClick', item.label)
    }
    const isMobile = () => {
      return window.outerWidth < 376
    }
    const menu = ref()
    const toggleMenu = (event: boolean) => {
      menu.value.toggle(event)
    }
    return {
      tabClick,
      isMobile,
      toggleMenu,
      menu,
      sm
    }
  }
})
</script>

<template>
  <span class="flex" :class="sm ? 'justify-between mb-5' : ''">
    <img src="/big3dlogo.png" class="w-16 h-16 md:w-24 md:h-24" />
    <div v-if="!sm" class="flex items-center overflow-x-auto">
      <div v-for="(item, index) in navItems" :key="index">
        <slot name="nav-item" :item="item">
          <slot :name="item.label" :item="item">
            <div
              class="nav-item link link-underline link-underline-black"
              @click="tabClick(item)"
            >
              {{ item.label }}
            </div>
          </slot>
        </slot>
      </div>
    </div>
    <span class="flex">
      <button
        class="icon-btn mx-2 !outline-none"
        title="button.toggle_dark"
        @click="toggleDark()"
      >
        <div i="carbon-sun dark:carbon-moon" />
      </button>
      <span
        v-if="sm"
        class="bg-yellow flex items-center justify-center h-16 w-24"
        @click="toggleMenu"
      >
        <img src="/icons/hamburgerMenu.svg" />
      </span>
    </span>
    <Menu ref="menu" :model="navItems" :popup="true" />
  </span>
</template>

<style scoped>
div:nth-child(n) {
  @apply p-2 w-full h-full;
}

.nav-item {
  @apply cursor-pointer text-xl;
}

.link-underline {
  border-bottom-width: 0;
  background-image: linear-gradient(transparent, transparent),
    linear-gradient(#fff, #fff);
  background-size: 0 3px;
  background-position: 0 100%;
  background-repeat: no-repeat;
  transition: background-size 0.5s ease-in-out;
}

.link-underline-black {
  background-image: linear-gradient(transparent, transparent),
    linear-gradient(#000, #000);
}

.link-underline:hover {
  background-size: 100% 3px;
  background-position: 0 100%;
}
</style>
