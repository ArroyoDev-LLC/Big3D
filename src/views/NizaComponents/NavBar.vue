<template>
  <span
    class="flex"
    :class="isMobile() ? 'bg-big3dBlack justify-between mb-5' : ''"
  >
    <span><img src="big3dlogo.png" /></span>
    <div class="flex flex-col sm:flex-row" v-if="!isMobile()">
      <div @click="tabClick('home')" class="cursor-pointer">Home</div>
      <div @click="tabClick('story')" class="cursor-pointer">Story+</div>
      <div @click="tabClick('samples')" class="cursor-pointer">Samples+</div>
      <div @click="tabClick('gallery')" class="cursor-pointer">Gallery</div>
      <div @click="tabClick('pricing')" class="cursor-pointer">Pricing</div>
      <div @click="tabClick('printing')" class="cursor-pointer">Printing</div>
      <div @click="tabClick('purchase')" class="cursor-pointer">Purchase</div>
      <div @click="tabClick('login')" class="bg-yellow cursor-pointer">
        Login
      </div>
    </div>
    <span
      v-if="isMobile()"
      class="bg-yellow flex items-center justify-center w-24"
      @click="toggleMenu"
    >
      <img src="/icons/hamburgerMenu.svg" />
    </span>
    <Menu :model="menuItems" ref="menu" :popup="true" />
  </span>
</template>

<script lang="ts">
import { defineComponent, ref } from "vue";
import Menu from "primevue/menu";

export default defineComponent({
  name: "NavBar",
  components: { Menu },
  setup(props, { emit }) {
    const tabClick = (section: string) => {
      emit("tabClick", section);
    };
    const isMobile = () => {
      return window.outerWidth < 376;
    };
    const menuItems = ref([
      {
        label: "Home",
        command: () => {
          tabClick("home");
        },
      },
      {
        label: "Story+",
        command: () => {
          tabClick("story");
        },
      },
      {
        label: "Samples+",
        command: () => {
          tabClick("samples");
        },
      },
      {
        label: "Gallery",
        command: () => {
          tabClick("gallery");
        },
      },
      {
        label: "Pricing",
        command: () => {
          tabClick("pricing");
        },
      },
      {
        label: "Printing",
        command: () => {
          tabClick("printing");
        },
      },
      {
        label: "Purchase",
        command: () => {
          tabClick("purchase");
        },
      },
      {
        label: "Login",
        command: () => {
          tabClick("login");
        },
      },
    ]);
    const menu = ref();
    const toggleMenu = (event) => {
      menu.value.toggle(event);
    };
    return {
      tabClick,
      isMobile,
      menuItems,
      toggleMenu,
      menu,
    };
  },
});
</script>

<style scoped>
.nav-container {
  @apply flex justify-evenly;
}

div:nth-child(n) {
  @apply p-6 w-full h-full;
}
</style>
