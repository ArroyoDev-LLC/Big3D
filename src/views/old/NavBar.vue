<script lang="ts">
import { defineComponent, ref } from 'vue'
import Menu from 'primevue/menu'
import { breakpointsTailwind, useBreakpoints } from '@vueuse/core'
const breakpoints = useBreakpoints(breakpointsTailwind)
const sm = breakpoints.smaller('sm')

export default defineComponent({
  name: 'NavBar',
  components: { Menu },
  props: {
    isUserLoggedIn: {
      type: Boolean,
      required: true
    },
    navItems: {
      type: Array<unknown>,
      default: []
    }
  },
  setup(props, { emit }) {
    const tabClick = (section: string) => {
      emit('tabClick', section)
    }
    const isMobile = () => {
      return window.outerWidth < 376
    }
    const menuItems = ref([
      {
        label: 'Home',
        command: () => {
          tabClick('home')
        }
      },
      {
        label: 'Story+',
        command: () => {
          tabClick('story')
        }
      },
      {
        label: 'Samples+',
        command: () => {
          tabClick('samples')
        }
      },
      {
        label: 'Gallery',
        command: () => {
          tabClick('gallery')
        }
      },
      {
        label: 'Pricing',
        command: () => {
          tabClick('pricing')
        }
      },
      {
        label: 'Purchase',
        command: () => {
          tabClick('generator-wizard')
        }
      },
      {
        label: 'Login',
        command: () => {
          tabClick('login')
          emit('login')
        }
      },
      {
        label: 'Account',
        command: () => {
          tabClick('account')
        }
      }
    ])
    const menu = ref()
    const toggleMenu = (event: boolean) => {
      menu.value.toggle(event)
    }
    return {
      tabClick,
      isMobile,
      menuItems,
      toggleMenu,
      menu,
      sm
    }
  }
})
</script>

<template>
  <span class="flex" :class="sm ? 'justify-between mb-5' : ''">
    <img src="/big3dlogo.png" class="w-16 h-16 md:w-24 md:h-24" />
    <div v-if="!sm" class="flex items-center overflow-x-auto">
      <div v-for="(item, index) in navItems" :key="index">
        <div
          v-if="item.label !== 'Login'"
          class="nav-item link link-underline link-underline-black"
          @click="item.command"
        >
          {{ item.label }}
        </div>
        <div v-else>
          <div
            v-if="!isUserLoggedIn"
            class="login-button"
            @click="item.command"
          >
            {{ item.label }}
          </div>
          <div
            v-else
            class="rounded-xl shadow-lg flex justify-center items-center"
            style="border: 1px solid black; height: 5rem"
            @click="$router.push('account')"
          >
            Triston
          </div>
        </div>
      </div>
    </div>
    <span class="flex">
      <button
        class="icon-btn mx-2 !outline-none"
        title="button.toggle_dark"
        @click="toggleDark()"
      >
        <div i="carbon-sun dark:carbon-moon" />
      </button>
      <span
        v-if="sm"
        class="bg-yellow flex items-center justify-center h-16 w-24"
        @click="toggleMenu"
      >
        <img src="/icons/hamburgerMenu.svg" />
      </span>
    </span>
    <Menu ref="menu" :model="navItems" :popup="true" />
  </span>
</template>

<style scoped>
.login-button {
  @apply bg-yellow text-xl hover:bg-gold hover:scale-105 hover:shadow-lg transform-gpu transition duration-300 ease-in-out cursor-pointer;
}

div:nth-child(n) {
  @apply p-2 w-full h-full;
}

.nav-item {
  @apply cursor-pointer text-xl;
}

.link-underline {
  border-bottom-width: 0;
  background-image: linear-gradient(transparent, transparent),
    linear-gradient(#fff, #fff);
  background-size: 0 3px;
  background-position: 0 100%;
  background-repeat: no-repeat;
  transition: background-size 0.5s ease-in-out;
}

.link-underline-black {
  background-image: linear-gradient(transparent, transparent),
    linear-gradient(#000, #000);
}

.link-underline:hover {
  background-size: 100% 3px;
  background-position: 0 100%;
}
</style>
