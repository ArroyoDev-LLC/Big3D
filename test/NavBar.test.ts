import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import NavBar from '../src/views/old/NavBar.vue'

describe('NavBar', () => {
  it('should render a nav bar', () => {
    const wrapper = mount(NavBar)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
