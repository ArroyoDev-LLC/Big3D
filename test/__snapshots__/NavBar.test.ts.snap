// Vitest Snapshot v1

exports[`NavBar > should render a nav bar 1`] = `
"<span class=\\"flex\\" data-v-36babc40=\\"\\"><img src=\\"/big3dlogo.png\\" class=\\"w-16 h-16 md:w-24 md:h-24\\" data-v-36babc40=\\"\\"><div class=\\"flex items-center overflow-x-auto\\" data-v-36babc40=\\"\\"><div class=\\"nav-item link link-underline link-underline-black\\" data-v-36babc40=\\"\\"> Home </div><div class=\\"nav-item link link-underline link-underline-black\\" data-v-36babc40=\\"\\"> Story+ </div><div class=\\"nav-item link link-underline link-underline-black\\" data-v-36babc40=\\"\\"> Samples+ </div><div class=\\"nav-item link link-underline link-underline-black\\" data-v-36babc40=\\"\\"> Gallery </div><div class=\\"nav-item link link-underline link-underline-black\\" data-v-36babc40=\\"\\"> Pricing </div><div class=\\"nav-item link link-underline link-underline-black\\" data-v-36babc40=\\"\\"> Purchase </div><div class=\\"login-button\\" data-v-36babc40=\\"\\"> Login </div></div><span class=\\"flex\\" data-v-36babc40=\\"\\"><button class=\\"icon-btn mx-2 !outline-none\\" title=\\"button.toggle_dark\\" data-v-36babc40=\\"\\"><div i=\\"carbon-sun dark:carbon-moon\\" data-v-36babc40=\\"\\"></div></button><!--v-if--></span>
<!----></span>"
`;
